tools/wake-format/emitter.cpp

```diff
@@ -25,10 +25,30 @@
 
 #include "parser/parser.h"
 
+#define ASSERT_TOKEN(node, token) \
+  assert(!node.empty());          \
+  assert(node.id() == token);
+
+#define APPEND_OR_BUBBLE(func, ctx, node) \
+  {                                       \
+    auto subtree = func(ctx, node);       \
+    if (!subtree) {                       \
+      return {};                          \
+    }                                     \
+    builder.append(*subtree);             \
```

Comment:
This assume `builder` is the name of the builder which is asking for trouble as soon as someone uses a different name. For instance you didn't do this with 'ctx' despite using that name consistently (probably because of ctx.flat() and such though).